### PR TITLE
Support more connections

### PR DESCRIPTION
**Pull request by egelja**
Complete rework of networking and also some cleanups and logging improvements

### CODE REVIEW

File: src/common.hpp [hunk 1]
```diff
@@ -9,22 +9,84 @@
...
 #include <cstring>
 
 #include <array>
+#include <memory>
 #include <string>
 #include <unordered_map>
 #include <utility>
 #include <vector>
 
+/***************************** Helper types **********************************/
```
Comment (echavemann) [inline src/common.hpp:22]: (for me) why do we need this?

File: src/main.cpp [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
+#include <curl/curl.h>
 #include <hiredis/hiredis.h>
 #include <quill/LogLevel.h>
+#include <uv.h>
 
 #include <iostream>
+#include <string_view>
 #include <tuple>
 
+constexpr static std::string_view COINBASE_SUBSCRIBE_STR = R"(
```
Comment (echavemann) [inline src/main.cpp:17]: where was this before lmao

File: src/main.cpp [hunk 3]
```diff
@@ -66,57 +94,63 @@ main(int argc, const char** argv)
...
+            [[unlikely]] // only the first message is like this
+        {
+            std::vector<uint8_t> bytes(
+                COINBASE_SUBSCRIBE_STR.begin(), COINBASE_SUBSCRIBE_STR.end()
+            );
             conn->send(std::move(bytes));
         }
-        else {
-            prox.process_incoming_data(string_data);
+        else [[likely]] {
```
Comment (echavemann) [inline src/main.cpp:142]: this keyword is sick wtf

**Review by echavemann — DISMISSED** [review]
lgtm very nce

File: src/common.hpp (hunk 1 above)
Comment (egelja) [reply src/common.hpp:22]: Look at the change in processing.cpp. It’s like passing Iterable in Python.

File: src/main.cpp (hunk 2 above)
Comment (egelja) [reply src/main.cpp:17]: In-line lmao.

File: src/main.cpp (hunk 3 above)
Comment (egelja) [reply src/main.cpp:142]: I know, right!? I was looking into using __builtin_likely and then found out about this.

**Review by stevenewald — DISMISSED** [review]
LGTM

**Comment by egelja** [comment]
Build actions passed I'm merging this

**Pull request merged**